app/src/org/commcare/activities/connect/ConnectIDManager.java

```diff
@@ -0,0 +1,565 @@
+
+import android.content.Intent;
+import android.widget.Toast;
+
+import org.commcare.activities.CommCareActivity;
+import org.commcare.android.database.connect.models.ConnectLinkedAppRecord;
+import org.commcare.android.database.connect.models.ConnectUserRecord;
+import org.commcare.core.network.AuthInfo;
+import org.commcare.preferences.AppManagerDeveloperPreferences;
+import org.javarosa.core.services.locale.Localization;
+
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Locale;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+public class ConnectIDManager {
+    public enum ConnectIDStatus {
+        NotIntroduced,
+        LoggedOut,
+        LoggedIn
+    }
+
+    public interface ConnectActivityCompleteListener {
+        void connectActivityComplete(boolean success);
+    }
+
+    private static ConnectIDManager manager = null;
+    private ConnectIDStatus connectStatus = ConnectIDStatus.NotIntroduced;
+    private CommCareActivity<?> parentActivity;
+    private ConnectActivityCompleteListener loginListener;
+    private int phase = ConnectIDConstants.CONNECT_NO_ACTIVITY;
+
+    //Only used for remembering the phone number between the first and second registration screens
+    private String primaryPhone = null;
+    private String recoveryPhone = null;
+    private String recoverySecret = null;
+    private boolean forgotPassword = false;
+
+    //Singleton, private constructor
+    private ConnectIDManager() {}
+
+    private static ConnectIDManager getInstance() {
+        if(manager == null) {
+            manager = new ConnectIDManager();
+        }
+
+        return manager;
+    }
+
+    public static void init(CommCareActivity<?> parent) {
+        ConnectIDManager manager = getInstance();
+        manager.parentActivity = parent;
+        ConnectIDDatabaseHelper.init(parent);
+
+        ConnectUserRecord user = ConnectIDDatabaseHelper.getUser(manager.parentActivity);
+        if(user != null && manager.connectStatus == ConnectIDStatus.NotIntroduced) {
+            manager.connectStatus = ConnectIDStatus.LoggedOut;
+        }
+    }
+
+    public static boolean isConnectIDIntroduced() {
+        return AppManagerDeveloperPreferences.isConnectIDEnabled() && getInstance().connectStatus != ConnectIDStatus.NotIntroduced;
+    }
+
+    public static boolean isSignedIn() {
+        return AppManagerDeveloperPreferences.isConnectIDEnabled() && getInstance().connectStatus == ConnectIDStatus.LoggedIn;
+    }
+
+    public static boolean shouldShowSignInMenuOption() {
+        return AppManagerDeveloperPreferences.isConnectIDEnabled() && getInstance().connectStatus == ConnectIDStatus.NotIntroduced;
+    }
+
+    public static boolean shouldShowSignOutMenuOption() {
+        return AppManagerDeveloperPreferences.isConnectIDEnabled() && getInstance().connectStatus == ConnectIDStatus.LoggedIn;
+    }
+
+    public static boolean isConnectIDActivity(int requestCode) {
```

Comment:
you should not need this method but can directly call `ConnectIDManager.handleFinishedActivity(requestCode, resultCode, intent)`, the `handleFinishedActivity` can return a boolean indicating whether it handled the activity response or not for calling activity to know whether the activity response was successfully handled by connect.